src/workflow/StatedWorkflow.js

```diff
@@ -83,6 +82,32 @@ export class StatedWorkflow {
         this.templateProcessor.functionGenerators.set("recoverStep", this.recoverStepGenerator.bind(this));
         this.templateProcessor.functionGenerators.set("subscribe", this.subscribeGenerator.bind(this));
         this.templateProcessor.logLevel = logLevel.ERROR; //log level must be ERROR by default. Do not commit code that sets this to DEBUG as a default
+        this.hasChanged = true;
+        this.changeListener = ()=>{this.hasChanged=true};
+        this.snapshotInterval = null;
+        //functions can be added here for all things that need to run during templateProcessor.onInitialize()
```

Comment:
We can alternatively use `rateLimit()` func like 
https://github.com/cisco-open/stated-workflow/blob/main/src/test/StatedWorkflow.test.js#L756-L759
```

    const dataChangeCallback = rateLimit(async (output, theseThatChanged) => {
        counts.push(output.interceptedMessages.length);
    }, snapshotIntervalSeconds * 1000);
    this.templateProcessor.setDataChangeCallback("/", dataChangeCallback);
```